docs/site/index/cldr-spec/picking-the-right-language-code.md

```diff
@@ -1,92 +1,92 @@
+	- For example, because "swh" occurs in \<languageAlias type\="swh" replacement\="sw"/\>, "sw" must be used instead of "swh".
+
+## Choosing Script/Territory Subtags
+
+If you need a particular variant of a language, then you'll add additional subtags, typically script or territory. Consult [Sample Subtags](http://unicode.org/cldr/utility/sample_subtags.html) for the most common choices. ***Again, review "[Caution](/index/cldr-spec/picking-the-right-language-code#caution)" section below.***
+
+## Verifying Your Choice
+
+1. Verify your choice by using the [online language identifier](http://unicode.org/cldr/utility/languageid.jsp) demo.
+2. You need to fix the identifier and try again in *any* if the demo shows any of the following:
+	1. the language identifer is illegal, or
+	2. one of the subtags is invalid, or
+	3. there are any replacement values.
+
+## Documenting Your Choice
+
+If you are requesting a new locale / language in CLDR, please include the links to the particular pages above so that we can process your request more quickly, as we have to double check before any addition. The links will be of the form:
+
+- https://iso639-3.sil.org/code/xxx
+- https://www.ethnologue.com/language/xxx
+- https://en.wikipedia.org/wiki/Luganda
+- and so on
+
+## Caution
+
+### Canonical Form
+
+Unicode language and locale IDs are based on BCP 47, but differ in a few ways. The canonical form is produced by using the canonicalization based on BCP 47 (thus changing iw → he, and zh\-yue → yue), plus a few other steps:
+
+1. Replacing the most prominent encompassed subtag by the macrolanguage (cmn → zh)
+2. Canonicalizing overlong 3 letter codes (eng\-840 → en\-US)
+3. Minimizing according to the likely subtag data (ru\-Cyrl → ru, en\-US → en).
+4. BCP 47 also provides for "variant subtags", such as [zh\-Latn\-pinyin](http://unicode.org/cldr/utility/languageid.jsp?a=zh-Latn-pinyin). When there are multiple variant subtags, the canonical format for Unicode language identifiers puts them in alphabetical order.
+
+Note that the CLDR likely subtag data is used to minimize scripts and regions, *not* the IANA Suppress\-Script. The latter had a much more constrained design goal, and is more limited.
+
+In some cases, systems (or companies) may have different conventions than the Preferred\-Values in BCP 47 \-\- such as those in the Replacement column in the the [online language identifier](http://unicode.org/cldr/utility/languageid.jsp) demo. For example, for backwards compatibility, "iw" is used with Java instead of "he" (Hebrew). When picking the right subtags, be aware of these compatibility issues. *If a target system uses a different canonical form for locale IDs than CLDR, the CLDR data needs to be processed by remapping its IDs to the target system's.*

```

Comment:
```suggestion
In some cases, systems (or companies) may have different conventions than the Preferred\-Values in BCP 47 \-\- such as those in the Replacement column in the the [online language identifier](http://unicode.org/cldr/utility/languageid.jsp) demo. For example, for backwards compatibility, "iw" is used with Java instead of "he" (Hebrew). Be aware of these compatibility issues when picking the subtags. *If a target system uses a different canonical form for locale IDs than CLDR, the CLDR data needs to be processed by remapping its IDs to those of the target system.*

```